src/Microsoft.DotNet.Darc/DarcLib/LocalLibGit2Client.cs

```diff
@@ -358,7 +358,28 @@ public async Task Push(
                 }
         };
 
-        repo.Network.Push(remote, branch.CanonicalName, pushOptions);
+        int attempts = 0;
```

Comment:
Did you consider to use the `ExponentialRetry` class?

Reply:
I looked at `RetryAfterHandler` and `RetryAllHandler`, but forgot this one exists, will use it instead

Reply:
too bad `ExponentialRetry` doesn't give you attempt number in the logging function, could change that for a small improvement